### PR TITLE
[Bug fix] Using loaded checkpoint with --do_predict (instead of random init)

### PR DESCRIPTION
**Pull request by ethanjperez**
Without this fix, I'm getting near-random validation performance for a trained model, and the validation performance differs per validation run. I think this happens since the `model` variable isn't set with the loaded checkpoint, so I'm using a randomly initialized model. Looking at the model activations, they differ each time I run evaluation (but they don't with this fix).

### CODE REVIEW

**Comment by ethanjperez** [comment]
Tagging @srush @nateraw from the original [Lightning GLUE PR](https://github.com/huggingface/transformers/pull/3290) to check I'm not missing something?

**Comment by nateraw** [comment]
I'll check this out later tonight! I'm on mobile so I've just looked at your commit quickly...looks like you're right. I know in the past I've instantiated the model then called `model.load_from_checkpoint(loaded_ckpt)` so what you've got probably gets the same job done. The benefit of doin it the way I just mentioned is that if you already have a model object available from training, you can just the best ckpt into that. Either way works though!

**Comment by nateraw** [comment]
That was fast :smile: Looks good to me!

**Comment by ethanjperez** [comment]
Thanks for checking :) I'm still not able to reproduce my in-training validation performance though with the --do_predict flag, any ideas? I'm getting identical validation accuracy on different runs now, but the accuracy is still near random

**Comment by nateraw** [comment]
@ethanjperez I just [checked the docs](https://pytorch-lightning.readthedocs.io/en/latest/weights_loading.html), and it looks like the way we were doing it originally was correct.

```python
model = MyLightingModule.load_from_checkpoint(PATH)
model.eval()
y_hat = model(x)
```

The way that I was explaining to do it would require you to use `torch.load` on the checkpoint path, which you would then pass to `model.load_from_checkpoint`. The above method (what we had originally) is probably supposed to do that for you.

I haven't had the chance to recreate the issue, so I'll have to take a look.

**Comment by ethanjperez** [comment]
Cool thanks! Even with the original way, I was still not able to reproduce my in-training validation performance (just something to look out for when you try) - In particular, I'm loading/running an already trained model  with the `--do_predict` flag without using the `--do_train` flag (I don't think you'd see the issue if you use both `--do_predict` and `--do_train`)

**Review by sshleifer — APPROVED** [review]
Great catch!

**Comment by ethanjperez** [comment]
@nateraw @sshleifer Are you guys able to load a trained model successfully with the pytorch-lightning scripts? Even after this patch, I am having issues loading an already trained model, i.e., if I just use `--do_eval` without also using `--do_train`

**Comment by sshleifer** [comment]
Sorry for taking so long. I will try to reproduce this today if there is no update on your end!

**Comment by ethanjperez** [comment]
@sshleifer Just seeing this - were you able to reproduce the issue? I can't remember what exact command I ran, but it was a standard evaluation command (the same as the training command I used, but with a few flags tweaked, e.g. drop the `--do-train` flag and add the `--do-eval` flag)

**Comment by sshleifer** [comment]
This is fixed now.
